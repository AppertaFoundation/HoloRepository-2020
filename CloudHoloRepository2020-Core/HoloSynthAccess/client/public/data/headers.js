export const headers =
[
    // {
    //   label: 'Series UID',
    //    field: 'SeriesInstanceUID',
    //    sort: 'asc'
    //
    // },
    {
        label: 'Date',
        field: 'SeriesDate',
        sort: 'asc'

    },
    {
        label: 'Description',
        field: 'SeriesDescription',
        sort: 'asc'

    },
    {
        label: 'BodyPart',
        field: 'BodyPartExamined',
        sort: 'asc'

    },
    {
        label: 'Collection',
        field: 'Collection',
        sort: 'asc'

    },
    {
        label: 'Download',
        field: 'download',
        sort: 'asc'

    }
]
