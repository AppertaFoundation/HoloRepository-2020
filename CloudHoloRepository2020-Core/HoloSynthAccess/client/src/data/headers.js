export const headers =
[

    {
        label: 'Date',
        field: 'SeriesDate',
        sort: 'asc'

    },
    {
        label: 'Description',
        field: 'SeriesDescription',
        sort: 'asc'

    },
    {
        label: 'BodyPart',
        field: 'BodyPartExamined',
        sort: 'asc'

    },
    {
        label: 'Collection',
        field: 'Collection',
        sort: 'asc'

    },
    {
        label: 'Image Count',
        field: 'ImageCount',
        sort: 'asc'

    },
    {
        label: '',
        field: 'download',
        sort: 'asc'

    }
]
