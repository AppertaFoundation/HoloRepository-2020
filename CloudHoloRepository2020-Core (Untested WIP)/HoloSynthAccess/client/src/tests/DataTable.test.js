import React from 'react';
import { shallow } from 'enzyme';
import DataTable from '../dataTable.js';
import renderer from 'react-test-renderer';
import testData from 'data/lung.json';
import { headers } from './data/headers.js';

const data = {
    bodyPart:'lung',
    local:false
}

let wrapper = shallow(<DataTable
    bodyPart={data.bodyPart}
    local={data.local}
/>);

let spyGetImagingStudies = jest
  .spyOn(DataTable.prototype, "dataFetch")
  .mockImplementation(() => Promise.resolve({
    columns: headers,
    rows: testData
  }));

afterAll(() => {
    spyGetImagingStudies.mockClear();
});

describe('App component', () => {
  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set the props', () => {
    expect(wrapper.prop("bodyPart")).toEqual("lung");
});

});