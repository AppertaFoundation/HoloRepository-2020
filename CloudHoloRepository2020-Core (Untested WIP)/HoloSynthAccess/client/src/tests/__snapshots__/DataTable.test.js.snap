// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders correctly 1`] = `
<div>
  <LoadingIndicator />
  <DataTable
    autoWidth={false}
    barReverse={false}
    bordered={true}
    borderless={false}
    btn={false}
    dark={false}
    data={Object {}}
    disableRetreatAfterSorting={false}
    displayEntries={true}
    entries={10}
    entriesLabel="Show entries"
    entriesOptions={
      Array [
        10,
        20,
        50,
        100,
      ]
    }
    exportToCSV={false}
    fixed={false}
    hover={false}
    info={true}
    infoLabel={
      Array [
        "Showing",
        "to",
        "of",
        "entries",
      ]
    }
    noBottomColumns={false}
    noRecordsFoundLabel="No matching records found"
    order={Array []}
    pagesAmount={8}
    paginationLabel={
      Array [
        "Previous",
        "Next",
      ]
    }
    paging={true}
    responsive={false}
    responsiveLg={false}
    responsiveMd={false}
    responsiveSm={false}
    responsiveXl={false}
    scrollX={false}
    scrollY={false}
    searchLabel="Search"
    searching={true}
    small={true}
    sortable={true}
    striped={true}
    tbodyColor=""
    tbodyTextWhite={false}
    theadColor=""
    theadTextWhite={false}
  />
</div>
`;
